src/diffusers/models/unet_2d_blocks.py

```diff
@@ -1384,6 +1462,195 @@ def forward(
+                    conv_shortcut_bias=False,
+                )
+            )
+            attentions.append(
+                BasicTransformerBlock(
+                   out_channels, 
+                   out_channels // attn_num_head_channels,
+                   attn_num_head_channels,
+                   dropout = None, 
+                   cross_attention_dim = cross_attention_dim,
+                   temb_channels = temb_channels,
+                   attention_bias = True,
+                   attn1_type = attn1_type,
+                   attn2_type = attn2_type,
+                   with_ff = False,
+                   use_conv_proj = True,
+                   cross_attention_norm = True,
+                   attention_dropout = dropout,
+                   group_size = resnet_group_size,
+                )
+            )
+
+        self.resnets = nn.ModuleList(resnets)
+        self.attentions = nn.ModuleList(attentions)
+
+        if add_downsample:
+            self.downsamplers = nn.ModuleList(
+                [KDownsample2d()]
+            )
+        else:
+            self.downsamplers = None
+        
+        self.gradient_checkpointing = False
+
+    def forward(self, hidden_states, temb=None, encoder_hidden_states=None, attention_mask=None, cross_attention_kwargs=None
+    ):
+        output_states = ()
+
+        if self.downsamplers is not None:
+            for downsampler in self.downsamplers:
+                hidden_states = downsampler(hidden_states)
+
+        for resnet, attn in zip(self.resnets, self.attentions):
+            if self.training and self.gradient_checkpointing:
+
+                def create_custom_forward(module, return_dict=None):
+                    def custom_forward(*inputs):
+                        if return_dict is not None:
+                            return module(*inputs, return_dict=return_dict)
+                        else:
+                            return module(*inputs)
+
+                    return custom_forward
+
+                hidden_states = torch.utils.checkpoint.checkpoint(create_custom_forward(resnet), hidden_states, temb)
+                hidden_states = torch.utils.checkpoint.checkpoint(
+                    create_custom_forward(attn, return_dict=False),
+                    hidden_states,
+                    encoder_hidden_states,
+                    attention_mask,
+                    cross_attention_kwargs,
+                )
+            else:
+                hidden_states = resnet(hidden_states, temb)
+                hidden_states = attn(
+                    hidden_states,
+                    encoder_hidden_states=encoder_hidden_states,
+                    emb = temb,
+                    attention_mask=attention_mask,
+                    cross_attention_kwargs=cross_attention_kwargs,)
+        
+        output_states += (hidden_states,)
+
+        return hidden_states, output_states
+
+
 class AttnUpBlock2D(nn.Module):
```

Comment:
Do we need this block for the new model? If not let's try to not change it :-)

Reply:
The changes should be reverted here since we don't need this block for the latent upscaling model

Reply:
will do!

It's not used in latent upscaler but I changed it because it also calls the `BasicTransformerBlock`, and I had replaced the `only_cross_attention` arguments in `BasicTransformerbLock` with `attn1_type` and `attn2_type` so it will break( should not have done that). We will refactor the `BasicTransformerBlock` soon so it should be fine